client/my-sites/stats/controller.jsx

```diff
@@ -378,5 +378,30 @@ module.exports = {
 				context.store
 			);
 		}
+	},
+
+	activity_log: function( context ) {
+		let siteId = context.params.site_id;
+		const site = getSite( context.store.getState(), siteId );
+		siteId = site ? ( site.ID || 0 ) : 0;
```

Comment:
Good find, let's work on that one in another PR though. This approach is already being used all throughout stats, so it makes complete sense to fix it in another PR.